packages/contracts/test/transactionManager.spec.ts

```diff
@@ -530,18 +559,109 @@ describe("TransactionManager", function () {
       ).to.be.revertedWith("prepare: SAME_CHAINIDS");
     });
 
-    it("should revert if param sending or receiving chainId doesn't match chainId variable", async () => {});
+    it("should revert if param sending or receiving chainId doesn't match chainId variable", async () => {
+      const { transaction, record } = await getTransactionData({ sendingChainId: 1340, receivingChainId: 1341 });
+      await expect(
+        transactionManager
+          .connect(user)
+          .prepare(transaction, record.amount, record.expiry, "0x", "0x", { value: record.amount }),
+      ).to.be.revertedWith("prepare: INVALID_CHAINIDS");
+    });
```

Comment:
Ideally we'd refactor the utils to make them rejectable

Reply:
gotchaa